sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala

```diff
@@ -3101,6 +3101,81 @@ abstract class DDLSuite extends QueryTest with SQLTestUtils {
       assert(spark.sessionState.catalog.isRegisteredFunction(rand))
     }
   }
+
+  test("SPARK-32481 Move data to trash on truncate table if enabled") {
+    val trashIntervalKey = "fs.trash.interval"
+    withTable("tab1") {
+      withSQLConf(SQLConf.TRUNCATE_TRASH_ENABLED.key -> "true") {
+        sql("CREATE TABLE tab1 (col INT) USING parquet")
+        sql("INSERT INTO tab1 SELECT 1")
+        // scalastyle:off hadoopconfiguration
+        val hadoopConf = spark.sparkContext.hadoopConfiguration
+        // scalastyle:on hadoopconfiguration
+        val originalValue = hadoopConf.get(trashIntervalKey, "0")
+        val tablePath = new Path(spark.sessionState.catalog
+          .getTableMetadata(TableIdentifier("tab1")).storage.locationUri.get)
+
+        val fs = tablePath.getFileSystem(hadoopConf)
+        val trashCurrent = new Path(fs.getHomeDirectory, ".Trash/Current")
```

Comment:
can we use `Trash.getCurrentTrashDir` instead? semantically it is the same but hides details such as homedir and others.

Reply:
is `Trash.getCurrentTrashDir` available for haddop2.7? I cannot find it

Reply:
Eh it is in 2.7.4 and up. Maybe you can use `Trash.getTrashPolicy().getCurrentTrashDir()` instead.

Reply:
I am afraid i couldn't find this as well :(  is it not recommended to use `getHomeDirectory` ? I see we have used it at some places before like [this](https://github.com/apache/spark/blob/2dee4352a0a3578818f1e364ee324fc012fbb91d/resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala#L187)

Reply:
Hmm okay. This is mostly a nit as this is only used in UT, so it's fine from my side.